include/kafka/BrokerMetadata.h

```diff
@@ -101,7 +113,6 @@ struct BrokerMetadata {
 
     void setOrigNodeName(const std::string& origNodeName)                          { _origNodeName = origNodeName; }
     void addNode(Node::Id nodeId, const Node::Host& host, Node::Port port)         { _nodes[nodeId] = std::make_shared<Node>(nodeId, host, port); }
-    std::shared_ptr<Node> getNode(Node::Id nodeId)                                 { return _nodes[nodeId]; }
```

Comment:
Is this the place where we insert nullptr shared_ptr?

Reply:
Exactly the place, --`getNode` returned the 'null', which was later inserted into the replica list. then it would crash while trying to iterate printing them for a partition.